### PR TITLE
finish import function

### PR DESCRIPTION
**Pull request by Roland-Hufnagel**

### CODE REVIEW

File: pages/dashboard/index.js [hunk 1]
```diff
@@ -1,7 +1,46 @@
...
+      const [word, translation] = entry.trim().split(";");
+      handleAddNewWord(word, translation);
+    });
+  }
+
+  function validateFileFormat(fileName) {
+    if (!fileName.endsWith(".csv")) {
+      alert(`${fileName} \n Format not supported! Please choose another file.`);
+    }
+  }
```
Comment (kai-franke) [inline pages/dashboard/index.js:42]:
```suggestion
  function validateFileFormat(fileName) {
    if (!fileName.endsWith(".csv")) {
      alert(`${fileName} \n Format not supported! Please choose another file.`);
      return false
    } else {
      return true
    }
  }

  function onChangeHandler(event) {
    const isValid = validateFileFormat(event.target.value);
    if (!isValid) {
      event.target.value = '';
    }
  }
```

File: pages/dashboard/index.js [hunk 2]
```diff
@@ -16,10 +55,70 @@ export default function Dashboard({ settings, handleChangeSettings }) {
...
+          {" "}
+          <form onSubmit={handleFileUpload}>
+            <StyledInput
+              id="file-upload"
+              type="file"
+              name="data"
+              required
+              onChange={(event) => {
+                validateFileFormat(event.target.value);
+              }}
```
Comment (kai-franke) [inline pages/dashboard/index.js:80]:
```suggestion
              onChange={(event) => onChangeHandler(event)}
```

File: pages/dashboard/index.js [hunk 3]
```diff
@@ -16,10 +55,70 @@ export default function Dashboard({ settings, handleChangeSettings }) {
           );
         })}
       </SettingsList>
+      <UploadCSV>
+        <CSVLabel>Dateiupload</CSVLabel>
+        <CSVDescription>
+          You can upload your own dictionary. Only CSV-files are supported.{" "}
```
Comment (kai-franke) [inline pages/dashboard/index.js:61]: Do you need `{" "}`?

File: pages/dashboard/index.js [hunk 4]
```diff
@@ -16,10 +55,70 @@ export default function Dashboard({ settings, handleChangeSettings }) {
...
+        </CSVDescription>{" "}
```
Comment (kai-franke) [inline pages/dashboard/index.js:62]: Here is another `{" "}`.

File: pages/dashboard/index.js [hunk 5]
```diff
@@ -16,10 +55,70 @@ export default function Dashboard({ settings, handleChangeSettings }) {
...
+        <Button
+          onClick={() => setShowUploadModal(true)}
+          text="Upload"
+          icon={EditIcon}
+        />
+      </UploadCSV>{" "}
+      {showUploadModal && (
+        <Modal handleClose={() => setShowUploadModal(false)}>
+          {" "}
```
Comment (kai-franke) [inline pages/dashboard/index.js:71]: And one more`{" "}`.

File: pages/dashboard/index.js [hunk 6]
```diff
@@ -16,10 +55,70 @@ export default function Dashboard({ settings, handleChangeSettings }) {
...
+        <CSVLabel>Dateiupload</CSVLabel>
+        <CSVDescription>
+          You can upload your own dictionary. Only CSV-files are supported.{" "}
+        </CSVDescription>{" "}
+        <Button
+          onClick={() => setShowUploadModal(true)}
+          text="Upload"
+          icon={EditIcon}
+        />
+      </UploadCSV>{" "}
```
Comment (kai-franke) [inline pages/dashboard/index.js:68]: I found another `{" "}`. 😀

File: pages/dashboard/index.js [hunk 7]
```diff
@@ -16,10 +55,70 @@ export default function Dashboard({ settings, handleChangeSettings }) {
...
     </>
   );
 }
 
+const StyledInput = styled.input`
+  &::file-selector-button {
+    padding: 0.5em;
+    border-radius: 5px;
+    background-color: var(--white);
+    padding: 0.35rem 0.75rem;
```
Comment (kai-franke) [inline pages/dashboard/index.js:95]: `padding` was defined twice.

**Review by kai-franke — COMMENTED** [review]
What a great feature and what a great job you did!

I've commented a suggestion for a small change to the file validation in order to avoid the upload of invalid file types. What do you think?

File: pages/dashboard/index.js (hunk 7 above)
Comment (Roland-Hufnagel) [reply pages/dashboard/index.js:95]: Ups

File: pages/dashboard/index.js (hunk 1 above)
Comment (Roland-Hufnagel) [reply pages/dashboard/index.js:42]: I like it very much 💪

**Review by kai-franke — APPROVED** [review]
Great! 👍

**Pull request merged**